lib/src/tasks/analyze/cli.dart

```diff
@@ -41,10 +43,18 @@ class AnalyzeCli extends TaskCli {
     bool fatalWarnings = TaskCli.valueOf(
         'fatal-warnings', parsedArgs, config.analyze.fatalWarnings);
     bool hints = TaskCli.valueOf('hints', parsedArgs, config.analyze.hints);
+    bool fatalHints = TaskCli.valueOf('fatal-hints', parsedArgs, config.analyze.failOnHints);
+
+    if (!hints && fatalHints) {
```

Comment:
might be better to implement hints as an option instead of a flag with values of 'none', 'enabled', 'fatal'?